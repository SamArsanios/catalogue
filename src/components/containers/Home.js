import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import SearchForm from './SearchForm';
import FilmContainer from './FilmContainer';
import Spinner from '../presentation/Spinner';

const Home = (props) => {
  const { loading } = props;

  return (
    <div className="container">
      <SearchForm />
      {loading ? <Spinner /> : <FilmContainer />}
    </div>
  );
};

const mapStateToProps = (state) => ({
  loading: state.films.loading,
});

Home.propTypes = {
  loading: PropTypes.bool,
};

Home.defaultProps = {
  loading: false,
};

export default connect(mapStateToProps)(Home);
