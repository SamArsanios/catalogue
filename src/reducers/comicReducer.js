import React from 'react'
import { FETCH_API_FAILURE } from '../actions/constants';

const initialState = {
  items: [],
  loading: false,
  error: null,
}

const comicReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_API_REQUEST:
      return {
        ...state,
        items: [],
        loading: true,
        error: null,
      };
    case FETCH_API_SUCCESS:
      return {
        ...state,
        items: action.comic,
        loading: false,
      };
    case FETCH_API_FAILURE:
      return {
        ...state,
        items: [],
        error: action.error,
        loading: false,
      }
    default:
      return state;
  }
}

export default comicReducer;
