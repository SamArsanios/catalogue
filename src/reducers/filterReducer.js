import { FILTER_FILM } from '../actions/constants';

export const initialState = {
  filter: 'All',
};

const filmFilterReducer = (state = initialState, action) => {
  switch (action.type) {
    case FILTER_FILM: {
      const filter = action.payload;
      return {
        ...state, filter,
      };
    }
    default:
      return { ...state };
  }
};

export default filmFilterReducer;
