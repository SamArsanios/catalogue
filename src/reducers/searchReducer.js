import { SEARCH_FILM, FETCH_FILMS } from '../actions/constants';

const initialState = {
  text: '',
  films: [],
  loading: false,
  film: [],
};

const searchReducer = (state = initialState, action) => {
  switch (action.type) {
    case SEARCH_FILM:
      return {
        ...state,
        text: action.payload,
        loading: false,
      };
    case FETCH_FILMS:
      return {
        ...state,
        films: action.payload,
      };
    default:
      return state;
  }
};

export default searchReducer;
