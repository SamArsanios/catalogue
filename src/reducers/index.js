import { applyMiddleware, combineReducers, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import searchReducer from './searchReducer';
import filmFilterReducer from './filterReducer';

const middleware = [thunk];
const initialState = {};

export const rootReducer = combineReducers({ films: searchReducer, filter: filmFilterReducer });

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware)),
);

export default store;
