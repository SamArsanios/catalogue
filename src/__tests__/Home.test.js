import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import Home from '../components/containers/Home';
import { rootReducer } from '../reducers/index';
import { composeWithDevTools } from 'redux-devtools-extension';

const state = {};
const middleware = [thunk];
const store = createStore(rootReducer, state, composeWithDevTools(applyMiddleware(...middleware)),);
const Wrapper = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);

Wrapper.propTypes = {
  children: PropTypes.shape({}).isRequired,
};

describe('Film', () => {
  it('renders correctly', () => {
    const tree = render(<Home />, { wrapper: Wrapper });
    expect(tree).toMatchSnapshot();
  });
});