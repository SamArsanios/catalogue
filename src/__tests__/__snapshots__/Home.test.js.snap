// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="jumbotron jumbotron-fluid mt-5 text-center"
        >
          <div
            class="container"
          >
            <h1
              class="display-4 mb-3"
            >
              <i
                class="fa fa-search"
              />
              Search Films
            </h1>
            <form
              id="searchForm"
            >
              <input
                class="form-control"
                name="searchText"
                placeholder="Search Films . . ."
                type="text"
              />
              <div>
                <select
                  aria-label="Default select example"
                  class="form-select my-5"
                >
                  <option
                    value="All"
                  >
                    All
                  </option>
                  <option
                    value="movie"
                  >
                    movie
                  </option>
                  <option
                    value="series"
                  >
                    series
                  </option>
                </select>
              </div>
              <button
                class="btn btn-dark w-25 btn-bg mb-3"
                type="submit"
              >
                Search
              </button>
            </form>
          </div>
        </div>
        <div
          class="row"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="jumbotron jumbotron-fluid mt-5 text-center"
      >
        <div
          class="container"
        >
          <h1
            class="display-4 mb-3"
          >
            <i
              class="fa fa-search"
            />
            Search Films
          </h1>
          <form
            id="searchForm"
          >
            <input
              class="form-control"
              name="searchText"
              placeholder="Search Films . . ."
              type="text"
            />
            <div>
              <select
                aria-label="Default select example"
                class="form-select my-5"
              >
                <option
                  value="All"
                >
                  All
                </option>
                <option
                  value="movie"
                >
                  movie
                </option>
                <option
                  value="series"
                >
                  series
                </option>
              </select>
            </div>
            <button
              class="btn btn-dark w-25 btn-bg mb-3"
              type="submit"
            >
              Search
            </button>
          </form>
        </div>
      </div>
      <div
        class="row"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
